src/entries/popup/pages/swap/SwapReviewSheet/SwapReviewSheet.tsx

```diff
@@ -243,6 +244,33 @@ const SwapReviewSheetWithQuote = ({
   const { minimumReceived, swappingRoute, includedFee, exchangeRate } =
     useSwapReviewDetails({ quote, assetToBuy, assetToSell });
 
+  const cleanNumber = (n: string): number => {
+    return parseFloat(n.replace(/,/g, ''));
+  };
+
+  const formatNumber = (n: string) =>
+    formatCurrency(cleanNumber(n), {
+      notation: 'compact',
+      maximumSignificantDigits: 4,
+    });
+
+  const processArray = (arr: string[]): string[] => {
+    return arr.map((item) => {
+      const parts = item.split(' ');
+      if (parts.length === 5) {
+        const formattedAmount = formatNumber(parts[3]);
+        return `${parts[0]} ${parts[1]} ${parts[2]} ${formattedAmount} ${parts[4]}`;
+      }
+      return item;
+    });
+  };
```

Comment:
I think this type of functionality should be dropped into a util file. Wdyt?

Reply:
yeah that sounds good to me!